pkg/eventsapi/generic_event.go

```diff
@@ -0,0 +1,48 @@
+package eventsapi
+
+import (
+	"encoding/json"
+)
+
+type GenericEvent struct {
```

Comment:
This structure feels a little strange at the moment since the name and capabilities make it _seem_ like an Event (e.g. it implements the `Event` interface).

Could we explore naming this something like `EventContainer` and maybe remove the implementations of `Event`?  It's possible we have a function like:

```go
func (ge GenericEvent) UnmarshalEvent() Event {
   // Deserialize the appropriate concrete type
   // Throw an error when there is no type match
}
```

and then the caller can use the `Event` interface of the returned type.

Reply:
Most of the changes are here: 00f0c6c